File: blacklist-backend/app/schemas/screening.py
"""
筛查相关数据模式
"""
from pydantic import BaseModel
from typing import Optional, List, Dict, Any
from datetime import datetime
from enum import Enum


class TaskStatus(str, Enum):
    """任务状态枚举"""
    PENDING = "pending"
    PROCESSING = "processing"
    COMPLETED = "completed"
    FAILED = "failed"


class MatchType(str, Enum):
    """匹配类型枚举"""
    PHONE = "phone"
    NAME = "name"
    KTT_NAME = "ktt_name"
    ADDRESS = "address"


class RiskLevel(str, Enum):
    """风险等级枚举"""
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"


class ScreeningTaskBase(BaseModel):
    """筛查任务基础模式"""
    task_name: str
    file_name: str
    file_path: Optional[str] = None


class ScreeningTaskCreate(ScreeningTaskBase):
    """创建筛查任务模式"""
    pass


class ScreeningTaskResponse(ScreeningTaskBase):
    """筛查任务响应模式"""
    id: int
    total_records: int = 0
    processed_records: int = 0
    matched_records: int = 0
    status: TaskStatus
    error_message: Optional[str] = None
    created_by: Optional[int] = None
    completed_at: Optional[str] = None
    is_active: bool
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True


class ScreeningResultResponse(BaseModel):
    """筛查结果响应模式"""
    id: int
    task_id: int
    blacklist_id: int
    order_data: Optional[Dict[str, Any]] = None
    match_type: MatchType
    match_score: Optional[float] = None
    match_details: Optional[str] = None
    risk_level: RiskLevel
    is_verified: str = "false"
    verified_by: Optional[int] = None
    verified_at: Optional[str] = None
    is_active: bool
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True

File: blacklist-frontend/src/api/screening.ts
/**
 * 筛查相关API
 */
import request from './request'

export interface ScreeningTask {
  id: number
  task_name: string
  file_name: string
  file_path: string
  total_records: number
  processed_records: number
  matched_records: number
  status: 'pending' | 'processing' | 'completed' | 'failed'
  error_message?: string
  created_by: number
  created_at: string
  completed_at?: string
}

export interface ScreeningResult {
  id: number
  task_id: number
  blacklist_id: number
  order_data: any
  match_type: 'phone' | 'name' | 'ktt_name' | 'address'
  match_score: number
  match_details: string
  risk_level: 'low' | 'medium' | 'high'
  created_at: string
}

export interface ScreeningTaskCreate {
  task_name?: string
}

export const screeningApi = {
  // 上传筛查文件
  uploadFile: (file: File, taskName?: string): Promise<ScreeningTask> => {
    const formData = new FormData()
    formData.append('file', file)
    if (taskName) {
      formData.append('task_name', taskName)
    }
    return request.post('/screening/upload', formData, {
      headers: {
        'Content-Type': 'multipart/form-data'
      }
    })
  },

  // 获取筛查任务列表
  getTasks: (params?: {
    skip?: number
    limit?: number
    status?: string
  }): Promise<ScreeningTask[]> => {
    return request.get('/screening/tasks', { params })
  },

  // 获取筛查任务详情
  getTask: (taskId: number): Promise<ScreeningTask> => {
    return request.get(`/screening/tasks/${taskId}`)
  },

  // 开始筛查任务
  startTask: (taskId: number): Promise<{ message: string; task_id: number }> => {
    return request.post(`/screening/tasks/${taskId}/start`)
  },

  // 获取筛查结果
  getResults: (taskId: number, params?: {
    skip?: number
    limit?: number
    risk_level?: string
  }): Promise<ScreeningResult[]> => {
    return request.get(`/screening/tasks/${taskId}/results`, { params })
  },

  // 导出筛查结果
  exportResults: (taskId: number, format: string = 'excel'): Promise<any> => {
    return request.post(`/screening/tasks/${taskId}/export`, null, {
      params: { format },
      responseType: 'blob'
    })
  }
}
